src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -192,6 +192,9 @@ private async Task ProcessRequestsWorker()
 
                     if (!Listener.ValidateRequest(requestContext))
                     {
+                        // Dispose the request
+                        requestContext.Dispose();
```

Comment:
This logic *could* be moved to ValidateRequest but it felt wrong to do it there. I'm fine either way..

Reply:
Should it also be disposed in the catch block in case ValidateRequest throws?

Reply:
Good catch